tools/validators/ontology_validator/yamlformat/tests/entity_type_manager_test.py

```diff
@@ -249,5 +250,67 @@ def testAnalyzeHandlesNamespaces(self):
     self.assertTrue(parent.HasFindingTypes(
         [findings_lib.DuplicateExpandedFieldSetsWarning]))
 
+  def testGetCompleteFieldSetsOI(self):
+    yaml = {'literals': ['field1', 'field2']}
+    field_universe = field_lib.FieldUniverse(
+        [_GetFieldFolder(yaml)])
+    yaml = {
+        'VAV_child1': {
+            'description': 'child1',
+            'uses': ['field1'],
+            'opt_uses': ['field2']
+        },
+    }
+    type_folder = _GetEntityTypeFolder(field_universe, yaml)
+    universe = _GetEntityTypeUniverse([type_folder])
+    manager = entity_type_manager.EntityTypeManager(universe)
+
+    self.assertRaises(Exception, manager.GetCompleteFieldSetsOI)
+
+    manager.Analyze()
+    function_output = manager.GetCompleteFieldSetsOI()
+
+    test_output = {
+        frozenset({
+            FieldParts(namespace='', field='field1', increment=''), 
+            FieldParts(namespace='', field='field2', increment='')
+        }): {'/VAV_child1'}
+    }
+
+    self.assertEqual(test_output, function_output)
+
+  def testGetTypenamesBySubsetOI(self):
```

Comment:
what does OI refers to?

Reply:
OI is part of how the mappings are defined in entity_type_manager.py. It means optionality insensitive. 
https://github.com/google/digitalbuildings/blob/89e4efb5be7f2305fb899c6252c1967027ebac6a/tools/validators/ontology_validator/yamlformat/validator/entity_type_manager.py#L82